osfoffline/gui/qt/preferences.py

```diff
@@ -84,12 +89,22 @@ def set_containing_folder(self):
             return
 
         user = Session().query(User).one()
+
+        if save_setting:
+            # copy the synced folders from the old to new location
+            # so OSF doesn't think they were deleted
+            copy_tree(user.folder, os.path.join(osf_path))
+
         user.folder = os.path.join(osf_path)
 
         self.containingFolderTextEdit.setText(self._translate("Preferences", self.containing_folder))
```

Comment:
Small aside: anything in the generated folder should probably be changed in QT Designer; otherwise the python-code changes might get lost next time someone made a change using QT Designer.